### ISSUE TITLE
gloo install fails with gloo-resource-rollout job not found error, using helm + tilt

### ISSUE THREAD

**Issue by jenshu** [solo-io/gloo#9033]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

v1.15.9

### Kubernetes Version

unknown

### Describe the bug

Customer is using helm and tilt to install GlooEE v1.15.9 (which uses Gloo OSS v1.15.17) and running into error:
```
Running cmd: ['helm', 'get', 'manifest', '--namespace', 'api-gateway', 'api-gateway']
Running cmd: ['kubectl', 'get', '-oyaml', '-f', '-']
Error from server (NotFound): jobs.batch "gloo-resource-rollout" not found
```
The error is thrown from helm-apply-helper [here](https://github.com/tilt-dev/tilt-extensions/blob/8381f694df7210551d523274f6336000ae934952/helm_resource/helm-apply-helper.py#L82)

In this version of GlooEE we have removed the helm hooks that used to be in the rollout job, and customer believes that is causing an ordering issue when trying to apply the resources. The rollout job and associated rbac resources did not get created.

As a workaround, they added back the helm hooks that were present in Gloo OSS v1.15.15 and the install succeeded.

### Expected Behavior

Install should succeed without workarounds.

### Steps to reproduce the bug

See above and check with customer.

**Comment by nfuden** [comment]
https://docs.tilt.dev/helm.html#pros-and-cons-of-helm

So basically if we want to support this behavior in Tilt we need to not use hooks 

Before digging in to far, should this be an install doc with a work around? Or is there a well known tilt startegy for handling hooks in general?

**Comment by davidjumani** [comment]
Have created a [test release](https://github.com/solo-io/solo-projects/actions/runs/7493661034) and awaiting user confirmation

**Comment by davidjumani** [comment]
Fixed in OSS [v1.17.0-beta5](https://github.com/solo-io/gloo/releases/tag/v1.17.0-beta5), [v1.16.3](https://github.com/solo-io/gloo/releases/tag/v1.16.3) && [v1.15.22](https://github.com/solo-io/gloo/releases/tag/v1.15.22)
This will be in Enterprise 1.17.0, 1.16.1, && 1.15.12
